src/ucp/proto/proto_single.c

```diff
@@ -29,21 +29,15 @@ ucp_proto_single_init_priv(const ucp_proto_single_init_params_t *params,
     ucp_lane_index_t lane;
     ucs_status_t status;
 
-    /* Slow path lanes are the last 3 (UCX_MAX_SLOW_PATH_LANES) lanes,
-     * so we need to shift them by 5 (UCX_MAX_FAST_PATH_LANES) */
-    ucp_lane_map_t slow_path_lanes = UCS_MASK(UCP_MAX_SLOW_PATH_LANES)
-                                     << UCP_MAX_FAST_PATH_LANES;
-
     num_lanes = ucp_proto_common_find_lanes(&params->super, params->lane_type,
-                                            params->tl_cap_flags, 1,
-                                            /* Exclude slow path lanes */
-                                            slow_path_lanes, &lane);
+                                            params->tl_cap_flags, 1, 0, &lane);
     if (num_lanes == 0) {
         ucs_trace("no lanes for %s", params->super.super.proto_name);
         return UCS_ERR_NO_ELEM;
     }
 
     ucs_assert(num_lanes == 1);
+    ucs_assert(lane < UCP_MAX_FAST_PATH_LANES);
```

Comment:
should document it here - lane_type of single protocols should cause only fast-paths lanes to be selected

Reply:
> IMO excluding slow path lanes were better, then this error handled by `num_lanes == 0` above. Otherwise it should be handled separately and treated as an error, warning or fatal.
@yosefe what was the reason you preferred to remove the exclude_map?